com.unity.render-pipelines.core/Editor/Utilities/MenuUtilities.cs

```
@@ -0,0 +1,159 @@
+
+            // add if required
+
+            // Note: GraphicsSettings.currentRenderPipeline == null possible and should mean built-in
+            // though we cannot use null as key so it was early replaced by typeof(RenderPipelineAsset)
+            Type currentSRPType = GraphicsSettings.currentRenderPipeline?.GetType() ?? typeof(RenderPipelineAsset);
+            if (!s_MenuMap.TryGetValue(currentSRPType, out var menuItemDatasForRP))
+                return;
+
+            foreach (var menuItemData in menuItemDatasForRP)
+                RegisterMenu(menuItemData);
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.Method)]
+    public class MenuItemForRenderPipeline : Attribute
+    {
+        public readonly Type[] renderPipelineAssetTypes;
+        public readonly string path;
+        public readonly int priority;
+        public readonly bool validate;
+
+        public MenuItemForRenderPipeline(string path, bool validate = false, int priority = 0, params Type[] renderPipelineAssetTypes)
+        {
+            if (renderPipelineAssetTypes == null || renderPipelineAssetTypes.Length == 0)
+                throw new ArgumentNullException("Argument cannot be null. Use MenuItem instead if you don't need to filter per RenderPipeline.", "renderPipelineTypes");
+
+            // Note: rpAssetType == null possible and should mean built-in
+            // but this is not possible as a key in s_MenuMap so we use the forbiden RenderPipelineAsset type for that
+            foreach (Type type in renderPipelineAssetTypes)
+                if (!typeof(RenderPipelineAsset).IsAssignableFrom(type) && type != null)
+                    throw new ArgumentException("Argument must be a RenderPipelineAsset child type", "renderPipelineTypes");
+
+            this.renderPipelineAssetTypes = renderPipelineAssetTypes
+                .Select(rpat => rpat != null ? rpat : typeof(RenderPipelineAsset)).ToArray();
```

I think we should throw an exception if the typeof is null or if the typeof is RenderPipelineAsset. This will mean invalid data. I think attribute must only be used to specify

---

This is only to support case where we want something only in builtin. Builtin is null (render pipeline asset used) but null cannot be used as a key